server/api/post_action.go

```diff
@@ -81,3 +82,49 @@ func (api *api) postActionRespond(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 }
+
+func (api *api) postActionConfirmStatusChange(w http.ResponseWriter, req *http.Request) {
+	mattermostUserID := req.Header.Get("Mattermost-User-ID")
+	if mattermostUserID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	response := model.PostActionIntegrationResponse{}
+	post := &model.Post{}
+
+	request := model.PostActionIntegrationRequestFromJson(req.Body)
+	if request == nil {
+		http.Error(w, "invalid request", http.StatusBadRequest)
+		return
+	}
+
+	value, ok := request.Context["value"].(bool)
+	if !ok {
+		http.Error(w, `No recognizable value for property "value"`, http.StatusBadRequest)
+		return
+	}
+
+	returnText := "The status has not been changed."
+	if value {
+		changeTo, ok := request.Context["change_to"]
+		if !ok {
+			http.Error(w, "No state to change", http.StatusBadRequest)
+			return
+		}
+		stringChangeTo := changeTo.(string)
```

Comment:
Nit: can you change this to match the type assertion strategy above?

Reply:
Sorry, forgot about that.